src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,57 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        if (robot.getX() != toX) {
+            moveRobotForX(robot, toX);
+        }
+        if (robot.getY() != toY) {
+            moveRobotForY(robot, toY);
+        }
+    }
+
+    private void moveRobotForY(Robot robot, int toY) {
+        if (robot.getY() < toY && !robot.getDirection().equals(Direction.UP)) {
+            turnRobot(robot, Direction.UP);
+        } else if (!robot.getDirection().equals(Direction.DOWN)) {
+            turnRobot(robot, Direction.DOWN);
+        }
+        while (robot.getY() != toY) {
+            robot.stepForward();
+        }
     }
-}
 
+    private void moveRobotForX(Robot robot, int toX) {
+        if (robot.getX() < toX && !robot.getDirection().equals(Direction.RIGHT)) {
+            turnRobot(robot, Direction.RIGHT);
+        } else if (!robot.getDirection().equals(Direction.LEFT)) {
+            turnRobot(robot, Direction.LEFT);
+        }
+        while (robot.getX() != toX) {
+            robot.stepForward();
+        }
+    }
+
+    private void turnRobot(Robot robot, Direction direction) {
+        if ((robot.getDirection().equals(Direction.LEFT) && direction.equals(Direction.RIGHT))
+                || (robot.getDirection().equals(Direction.RIGHT)
+                && direction.equals(Direction.LEFT))
+                || (robot.getDirection().equals(Direction.UP)
+                && direction.equals(Direction.DOWN))
+                || (robot.getDirection().equals(Direction.DOWN)
+                && direction.equals(Direction.UP))) {
+            robot.turnLeft();
+            robot.turnLeft();
+        } else if ((direction.equals(Direction.LEFT)
+                && robot.getDirection().equals(Direction.UP))
+                || (direction.equals(Direction.RIGHT)
+                && robot.getDirection().equals(Direction.DOWN))
+                || (direction.equals(Direction.UP)
+                && robot.getDirection().equals(Direction.RIGHT))
+                || (direction.equals(Direction.DOWN)
+                && robot.getDirection().equals(Direction.LEFT))) {
+            robot.turnLeft();
+        } else {
+            robot.turnRight();
+        }
+    }
```

Comment:
Try to optimize it.

Reply:
I did not understand the choice of turning right or left. I wanted to make a smart robot, and not just make it spin in place.